### ISSUE TITLE
Parallelizable and OneTimeSetup are not working together

### ISSUE THREAD

**Issue by HemalathaSR** [nunit/nunit#2440]
I am using OneTimeSetup to create a log folder, something like this:
```
[SetUpFixture]
    public class MySetUpClass
    {
        [OneTimeSetUp]
        public void RunBeforeAnyTests()
        {

            string folder = "C:\\Users\\" + Environment.UserName + "\\Desktop\\Automation\\" + DateTime.Now.ToString("yyyy_MM_dd_HH_mm_ss");
            if (!System.IO.Directory.Exists(folder))
            {
                System.IO.Directory.CreateDirectory(folder);
            }
        }
}
```

I have close to 15 tests in [TestFixture] which are suppose to run in Parallel, I went with [Parallelizable(ParallelScope.All)].

At a time I have 5 or 6 tests run in parallel but usually after around 10 tests, nunit hangs. Sometimes it hangs as soon as I start.

If I remove the OneTimeSetup and create a folder on my own, this doesn't happen. If I remove the Parallelizable, this doesn't happen.

How should I fix this?

**Comment by rprouse** [comment]
Are you sure it is the OneTimeSetup that is causing this? If your tests are running, it seems more likely that something in your tests is not parallizable. You say the folder is a log folder. I assume you are logging out to that folder from the tests. Is your logging thread safe?

**Comment by ChrisMaddock** [comment]
I'm suspicious this might relate to https://github.com/nunit/nunit/issues/2438 - which I've been looking at this afternoon. Just typing up some thoughts on that one now.

**Comment by ChrisMaddock** [comment]
@HemalathaSR - will you try the fix that's been put into this package, and see if that solves your issue?

https://ci.appveyor.com/project/CharliePoole/nunit/build/3.9.0-ci-04451-pr-2445/artifacts

**Comment by CharliePoole** [comment]
I've given myself the task of reviewing all our issues related to parallelism for possible incorporation in a PR I'm starting. This one, however, doesn't seem to have enough information.

@HemalathaSR Can you please respond to the questions from @rprouse and @ChrisMaddock? Otherwise, we don't seem to have enough info to go on in order to work the problem.

**Comment by ChrisMaddock** [comment]
This may have been fixed in 3.9, but there's not enough information here to know. Closing - happy to reopen if there's more to look at.
